src/Weather/MapOverlay/EdlControlsModel.cpp

```
@@ -46,35 +51,110 @@ EdlControlsModel::FillForecastChoices(DataFieldEnum &field) noexcept
+  if (HasOverlayData())
+    text = base;
+  else
+    WeatherMapOverlay::AppendNoDataTag(text, base.c_str());
+}
+
+void
+EdlControlsModel::FormatLevelLabel(StaticString<64> &text) const noexcept
+{
+  const unsigned isobar = EDL::GetIsobar();
+  const int altitude = EDL::GetAltitudeForIsobar(isobar);
+
```

Cppcheck (2.21.0)

[error] Dereference of an invalid iterator

(derefInvalidIterator)